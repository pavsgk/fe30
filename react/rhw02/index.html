<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>

<script>
  const sendReq = async () => {
    const result = await fetch('https://script.google.com/macros/s/AKfycbzEYFV5JBtD5d2D2swAs6i9n0SeqMxAgZRD-tJG6F3ONbSgF38kL4qeuJ-U-FMDebvxew/exec?___testS1hop1').then(res => res.json());
    console.log(result);
    return result;
  }

  sendReq();



</script>  
</body>
</html>