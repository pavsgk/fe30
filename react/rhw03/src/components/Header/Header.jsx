import { NavLink } from "react-router-dom";
import styles from "./Header.module.scss";

function Header () {

  return (
    <header className={styles.Header}>
      <NavLink to="/"> Shop</NavLink>
      <NavLink to="/cart">Cart</NavLink>
      <NavLink to="/fav">Favorites</NavLink>
    </header>
  )
}

export default Header;