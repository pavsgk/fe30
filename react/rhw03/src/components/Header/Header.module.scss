.Header {
  display: flex;
  justify-content: center;
  gap: 50px;
  align-items: center;
  width: 100%;
  height: 50px;
  background-color: #7CB9E8;
  
  a {
    color: #FFF;
  }
}