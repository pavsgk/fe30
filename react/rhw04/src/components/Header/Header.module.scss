.Header {
  display: flex;
  position: fixed;
  z-index: 999;
  top: 0;
  justify-content: center;
  gap: 50px;
  align-items: center;
  width: 100%;
  height: 50px;
  background-color: #7CB9E8;
  
  a {
    color: #FFF;
  }
}

.HeaderDummy {
  width: 100%;
  height: 50px;
  margin-bottom: 20px;
}