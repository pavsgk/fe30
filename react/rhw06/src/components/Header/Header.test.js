import Header from "./Header";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("Snapshot test", () => {
  test("Should component render", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
