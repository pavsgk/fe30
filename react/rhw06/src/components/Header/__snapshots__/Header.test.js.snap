// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Should component render 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
       Shop
    </a>
    <a
      href="/fav"
    >
      Favorites
    </a>
    <a
      href="/cart"
    >
      Cart
    </a>
  </header>
  <div
    class="HeaderDummy"
  />
</DocumentFragment>
`;
